### PR TITLE
ignore failed state when stopping units

### LINKED ISSUE #206: On updating, we should ignore failed units
On updating, when we remove units, we should ignore units with a failed state - they're going away anyway.

### PR DESCRIPTION
**Pull request by xh3b4sd**
Fixes #206 

RFR @JosephSalisbury

### CODE REVIEW

File: controller/status.go [hunk 1]
```diff
@@ -309,19 +309,25 @@ func (a Aggregator) AggregateStatus(fc, fd, sa, ss string) (Status, error) {
 	return aggregatedStatuses[0], nil
 }
 
-func (a Aggregator) unitHasStatus(us fleet.UnitStatus, status Status) (bool, error) {
+func (a Aggregator) UnitHasStatus(us fleet.UnitStatus, statuses ...Status) (bool, error) {
```
Comment (JosephSalisbury) [inline controller/status.go:315]: This could use a godoc

File: controller/update.go [hunk 2]
```diff
@@ -300,11 +300,16 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
...
-			// add
-			maxGrowth := opts.MaxGrowth + numTotal - opts.MinAlive - int(addInProgress)
+
+			currentSliceReq := req
+			// Calculating the number of groups allowed to be added during this
+			// iteration respects the total number of running groups. This is because
+			// we later going to check how many are actually running after the
+			// addition. So the next iteration will calculate and check again and
+			// respect the total number of running groups. See also
+			// isGroupAdditionAllowed.
```
Comment (JosephSalisbury) [inline controller/update.go:320]: I wouldn't mind seeing a similar explanation later, for the `minAlive` calculation - particularly why the two are different. This caught us out earlier.

**Comment by JosephSalisbury** [comment]
LGTM, few comments

File: controller/update.go (hunk 2 above)
Comment (xh3b4sd) [inline controller/update.go:320]: The calculation for `minAlive` is more straight forward. I would keep it as it is. There was no problem. I would not know what to document there besides the rather obvious information.

Comment (JosephSalisbury) [inline controller/update.go:320]: Works for me <3

**Pull request merged**